File: solutions/kotlin/difference-of-squares/2/src/main/kotlin/DifferenceOfSquares.kt
class Squares(private val number: Int) {
    private val sumOfSquares = (1..number).map{ it * it }.sum()
    private val sum = (1..number).sum()
    private val squareOfSum = sum * sum
    private val difference = squareOfSum - sumOfSquares
    
    fun sumOfSquares(): Int {
        return sumOfSquares
    }

    fun squareOfSum(): Int {
        return squareOfSum
    }

    fun difference(): Int {
        return difference
    }
}

File: solutions/go/difference-of-squares/2/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) int {
	switch {
    case n == 0:
        return 0
    case n > 0:
        s := 0
        for i := 1; i <= n; i++ {
            s += i
        }
        return s * s
    default:
        return SquareOfSum(-n)
    }
}

func SumOfSquares(n int) int {
	switch {
    case n == 0:
        return 0
    case n > 0:
        s := 0
        for i := 1; i <= n; i++ {
            s += i * i
        }
        return s
    default:
        return SumOfSquares(-n)
    }
}

func Difference(n int) int {
	diff := SquareOfSum(n) - SumOfSquares(n)
    switch {
    case diff < 0:
        return -diff
    default:
        return diff
    }
}
